src/index.ts

```diff
@@ -216,6 +216,10 @@ export default function typescript(options?: Partial<IOptions>)
 				const map = _.find(output.outputFiles, (entry) => _.endsWith(entry.name, ".map"));
 				const dts = _.find(output.outputFiles, (entry) => _.endsWith(entry.name, ".d.ts"));
 
+				if (pluginOptions.sourceMapCallback && map) {
```

Comment:
If you move this down to line 256 or so and grab map out of `result.map`, it will work with `clean: false`. Might have to save map.text in a separate property on `result` if you must have it as a text instead of parsed object.

Treat it similar to `dts` property if you need to do that.

Reply:
You can use result.map directly if you remove JSON.parse in instrument plugin

Reply:
* I clearly did not spend enough time to try and understand the transform function in full - makes sense now, thanks. 👍 
* There is a very good reason for handing the callback the JSON string rather than a reference to the `map` object, as I've learned the hard way: Rollup will mutate the object (by decoding the base64 VLQ mappings), so by the time the intrumenter gets hold of it, it's no longer usable.

I'll update the PR shortly.